doc/releases/changelog-dev.md

```diff
@@ -582,6 +583,9 @@
 * `Tensor._pauli_rep` is set to `None` during initialization. Add `Tensor.data` setter.
   [(#3722)](https://github.com/PennyLaneAI/pennylane/pull/3722)
 
+* Redirect `qml.math.ndim` to `jnp.ndim` when using it on a jax tensor.
```

Comment:
Was this a bug? What happened before? Shouldn't there be a new test then :thinking:

Reply:
It was a bug. `qml.math.ndim` was always redirected to `np.ndim`, and when calling it on a tracer an error would be raised.

Reply:
Oh dear! Thanks for catching this!